### PR TITLE
Rework `AccuracyCircle` to not use `BufferedContainer`s

### PR DESCRIPTION
**Pull request by EVAST9919**
Nothing groundbreaking: slightly better performance, a bit cleaner visually.

|master|pr|
|---|---|
|master-solo|pr-solo|
|master-results|pr-results|

### CODE REVIEW

File: osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs [hunk 1]
```diff
@@ -63,7 +62,7 @@ public partial class AccuracyCircle : CompositeDrawable
         /// <summary>
         /// Relative width of the rank circles.
         /// </summary>
-        public const float RANK_CIRCLE_RADIUS = 0.06f;
+        public const float RANK_CIRCLE_RADIUS = 0.05f;
```
Comment (peppy) [inline osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs:65]: Are these constant changes to account for the sharper display without buffered container, or something else?

Comment (EVAST9919) [reply osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs:65]: > Are these constant changes to account for the sharper display without buffered container

Basically yes. Without `BufferedContainer` they became a tiny bit thicker, so I'm just accounting for that. Used master as a visual reference. Can upload an image comparison mster/old values/new values

File: osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs [hunk 2]
```diff
@@ -74,12 +74,12 @@ public partial class AccuracyCircle : CompositeDrawable
...
         /// </summary>
-        private const double virtual_ss_percentage = 0.01;
+        public const double VIRTUAL_SS_PERCENTAGE = 0.01;
 
         /// <summary>
-        /// The width of a <see cref="RankNotch"/> in terms of accuracy.
+        /// The width of spacing in terms of accuracy between the grade circles.
         /// </summary>
-        public const double NOTCH_WIDTH_PERCENTAGE = 1.0 / 360;
+        public const double GRADE_SPACING_PERCENTAGE = 2.0 / 360;
```
Comment (bdach) [inline osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs:82]: Why has this increased two-fold too?

File: osu.Game/Screens/Ranking/Expanded/Accuracy/GradedCircles.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            get => progress;
+            set
+            {
+                progress = value;
+                dProgress.RevealProgress = value;
+                cProgress.RevealProgress = value;
+                bProgress.RevealProgress = value;
+                aProgress.RevealProgress = value;
+                sProgress.RevealProgress = value;
+                xProgress.RevealProgress = value;
```
Comment (bdach) [inline osu.Game/Screens/Ranking/Expanded/Accuracy/GradedCircles.cs:28]: This is pretty terrible. Can you put the `GradedCircle` instances in a `Container<GradedCircle>` and iterate over that instead?

File: osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs:82]: It's likely going to be for the same reason as https://github.com/ppy/osu/pull/26769#discussion_r1469654687.

Comment (EVAST9919) [reply osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs:82]: This one is different. Previously we were drawing boxes with `EdgeSmoothness = 1` on top of a circle, which resulted in visible spacing being approx `const + 2px`. Now it's the opposite: antialiasing is being handled by the circles and the visible spacing would be `const - 2px`. And to account for that I increased the const

**Review by bdach — APPROVED** [review]
Seems okay on a visual check

**Pull request merged**